cypress/integration/login.ts

```diff
@@ -0,0 +1,20 @@
+import { Web3Provider } from '@ethersproject/providers';
+
+import { TestProvider } from '../util';
+
+context('Coordinape', () => {
+  before(() => {
+    cy.on('window:before:load', win => {
+      const provider = new Web3Provider(
+        new TestProvider('http://localhost:8545')
+      );
+      Object.defineProperty(win, 'ethereum', { value: provider });
+    });
+  });
+  it('can login', () => {
+    cy.visit('http://localhost:3000');
```

Comment:
hmmmmm... oops, wasn't thinking about this until now. if cypress needs the app to be running, then for CI, that's another piece of the stack we'll have to setup & teardown along with the testchain & hasura.

Reply:
now i'm rethinking what i was saying earlier about prioritizing getting this into the CI step along with everything else... wondering if it's going to be too heavyweight.

Reply:
If only we could use the test hasura/vercel with simple seeding?

Reply:
We definitely can. But I'm talking about hosting the web app. Ordinary Jest tests don't need this because components are rendered in a virtual DOM

Reply:
that's what we'd want to do, yeah, it's just a bit of a devops boondoggle to get that all running both in dev & in a Github Action. my ideal goal here is that let you just run `yarn test:ci` in any environment, and everything would spin up with fresh data before running Jest tests and get cleaned up afterward.

i got most of the way through the Hasura side of it today, but ofc due to the auth webhook, vercel needs to run as well. i wonder if there's a way to spin up a lightweight process just for the auth hook...